src/main.ts

```diff
@@ -50,11 +50,15 @@ async function run(): Promise<undefined> {
       header
     )
 
-    if (!previous) {
+    if (!previous && !deleteOldComment) {
       await createComment(octokit, repo, pullRequestNumber, body, header)
       return
     }
 
+    if (!previous) {
+      return
+    }
+
     if (deleteOldComment) {
```

Comment:
@fddayan Can you move `deleteOldComment` check before of previous check?
I think It will be easier to read then spread deleteOldComment flag.

```js
    if (deleteOldComment) {
       if (previous) {
           ...
       }
       return
    }

    if (!previous) {
```